src/rt-api/post-contact.ts

```diff
@@ -90,42 +119,78 @@ export const postContact = async (
     });
   }
 
-  if (contact.contactFieldsToDelete){
-    console.log(contact.contactFieldsToDelete)
-    if(contact.contactFieldsToDelete.emailAddresses){
-      for(const internetRecId of contact.contactFieldsToDelete.emailAddresses){
-        const internetResult = await Axios({
-          method: "post",
+  // If present, update contact's url addresses
+  if (contact.Urls) {
+    contact.Urls.forEach(async (url: UrlUpdate) => {
+      try {
+        const method = url.id === 0 ? "post" : "put";
+        const urlUrl =
+          url.id === 0
+            ? `/contacts/${contact.ContactRecord.id}/urls`
+            : `/contacts/${contact.ContactRecord.id}/urls/${url.id}`;
+        if (url.id === 0) {
+          delete url.id;
+        }
+
+        const urlResult = await Axios({
+          method,
+          url: REDTAIL_TWAPI_URL + urlUrl,
+          headers: createRtApiHeaders(userKey),
+          data: url,
+        });
+        if (urlResult.status !== 200) {
+          return 5;
+        }
+      } catch (e) {
+        logger.error("Url RT API PUT error: " + JSON.stringify(e));
+        return 5;
+      }
+    });
+  }
+
+  if (contact.contactFieldsToDelete) {
+    if (contact.contactFieldsToDelete.addresses) {
+      for (const addressId of contact.contactFieldsToDelete.addresses) {
+        const addressResult = await Axios({
+          method: "delete",
           url:
-            REDTAIL_API_URL +
-            `/contacts/${contact.ContactRecord.ClientID}/internets/${internetRecId}?Type=email`,
+            REDTAIL_TWAPI_URL +
+            `/contacts/${contact.ContactRecord.id}/addresses/${addressId}`,
           headers: createRtApiHeaders(userKey),
         });
-        console.log(internetResult.status)
       }
     }
-    if(contact.contactFieldsToDelete.phoneNumbers){
-      for(const phoneRecId of contact.contactFieldsToDelete.phoneNumbers){
+    if (contact.contactFieldsToDelete.emails) {
+      for (const emailId of contact.contactFieldsToDelete.emails) {
+        const emailResult = await Axios({
+          method: "delete",
+          url:
+            REDTAIL_TWAPI_URL +
+            `/contacts/${contact.ContactRecord.id}/emails/${emailId}`,
+          headers: createRtApiHeaders(userKey),
+        });
+      }
+    }
```

Comment:
Same here I wonder if there's a way we can extract these delete calls into it's own function